pipeline/trusted-resources/pkg/trustedtask/trustedtaskrun.go

```diff
@@ -136,8 +131,27 @@ func prepareTask(t v1beta1.TaskObject) (v1beta1.Task, []byte, error) {
 		APIVersion: "tekton.dev/v1beta1",
 		Kind:       "Task"}
 	task.Spec = t.TaskSpec()
-	task.ObjectMeta = t.TaskMetadata()
 
+	// exclude the fields populated by system.
+	task.Name = t.TaskMetadata().Name
+	task.GenerateName = t.TaskMetadata().GenerateName
+	task.Namespace = t.TaskMetadata().Namespace
+
+	task.Labels = make(map[string]string)
```

Comment:
In particular, do we need to be sensitive to nil vs empty maps here?

Reply:
you mean check if t.TaskMetadata().Labels is nil ?

Reply:
I guess it is unnecessary?  because I don't use labels in my test and examples and they can work 😂  maybe I'm miss something

Reply:
I changed it, not sure if that makes sense